backend/app/vending/__init__.py

```diff
@@ -191,6 +260,201 @@ def get_dashboard_link(login=Depends(login_state)) -> VendingRedirect:
         raise VendingError("stripe-link-create-failed") from error
 
 
+@router.get("/config")
+def get_global_vending_config() -> VendingConfig:
+    """
+    Retrieve the configuration values needed to calculate application
+    vending splits client-side.
+
+    Configuration includes:
+    - Fee values
+    - Platform values
+    """
+    (
+        fee_fixed_cost,
+        fee_cost_percent,
+        fee_prefer_percent,
+    ) = prices.flathub_fee_parameters("usd")
+
+    return VendingConfig(
+        status="ok",
+        platforms=PLATFORMS,
+        fee_fixed_cost=fee_fixed_cost,
+        fee_cost_percent=fee_cost_percent,
+        fee_prefer_percent=fee_prefer_percent,
+    )
+
+
+@router.get("app/{appid}")
+def get_app_vending_status(appid: str) -> VendingDescriptor:
+    """
+    Retrieve the vending status for the given application.  Returns an error if
+    the appid is not known, or is not set up for vending.
+    """
+    vend = ApplicationVendingConfig.by_appid(db, appid)
+    if not vend:
+        return Response(status_code=204)
+    try:
+        shares = prices.compute_shares(appid, vend.appshare)
+    except ValueError as val_err:
+        raise VendingError(error="bad-app-share") from val_err
+
+    (
+        fee_fixed_cost,
+        fee_cost_percent,
+        fee_prefer_percent,
+    ) = prices.flathub_fee_parameters(vend.currency)
+
+    return VendingDescriptor(
+        status="ok",
+        currency=vend.currency,
+        components=shares,
+        fee_fixed_cost=fee_fixed_cost,
+        fee_cost_percent=fee_cost_percent,
+        fee_prefer_percent=fee_prefer_percent,
+    )
```

Comment:
Slight problem with this endpoint is that it can't be used to get previews in the developer setup part of frontend, because the app share itself could be changing.

I've boldly added a new endpoint (`@router.get("/config")`) to return the platform data and the fee data given here, that way I can just run the entire breakdown process client side. This seems to be the best approach I can think of in terms of both performance (can be fetched once statically) and UX (keeping a live responsive preview), but I'm not set on it if we don't like that.

Reply:
As discussed at sync: I've now removed the unused endpoints